Documentation/gettingstarted/cli-download.rst

```diff
@@ -8,30 +8,25 @@ various features (e.g. clustermesh, Hubble).
     .. code-block:: shell-session
 
       CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-linux-amd64.tar.gz{,.sha256sum}
-      sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
-      sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
-      rm cilium-linux-amd64.tar.gz{,.sha256sum}
+      CLI_ARCH=$(arch)
+      if [ "${CLI_ARCH}" = "x86_64" ] || [ "${CLI_ARCH}" = "i386" ]; then CLI_ARCH=amd64; fi
+      if [ "${CLI_ARCH}" = "aarch64" ]; then CLI_ARCH=arm64; fi
+      curl -L --fail --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
+      sha256sum --check cilium-linux-${CLI_ARCH}.tar.gz.sha256sum
+      sudo tar xzvfC cilium-linux-${CLI_ARCH}.tar.gz /usr/local/bin
+      rm cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
 
-  .. group-tab:: macOS (Intel)
+  .. group-tab:: macOS
 
     .. code-block:: shell-session
 
       CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-darwin-amd64.tar.gz{,.sha256sum}
-      shasum -a 256 -c cilium-darwin-amd64.tar.gz.sha256sum
-      sudo tar xzvfC cilium-darwin-amd64.tar.gz /usr/local/bin
-      rm cilium-darwin-amd64.tar.gz{,.sha256sum}
-
-  .. group-tab:: macOS (Apple Silicon - ARM)
-
-    .. code-block:: shell-session
-
-      CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-darwin-arm64.tar.gz{,.sha256sum}
-      shasum -a 256 -c cilium-darwin-arm64.tar.gz.sha256sum
-      sudo tar xzvfC cilium-darwin-arm64.tar.gz /usr/local/bin
-      rm cilium-darwin-arm64.tar.gz{,.sha256sum}
+      CLI_ARCH=$(arch)
```

Comment:
Will this determine the correct arch when the shell and/or terminal is an universal binary? See https://github.com/cilium/cilium/pull/19009#pullrequestreview-898661284 for previous discussion.

Reply:
Yeah, `arch` returns i386 when running under Rosetta, but `uname -m` returns arm64 as expected. This makes sense to me.